pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -1065,7 +1066,17 @@ public boolean trackDelayedDelivery(long ledgerId, long entryId, MessageMetadata
             delayedDeliveryTracker.get().resetTickTime(topic.getDelayedDeliveryTickTimeMillis());
 
             long deliverAtTime = msgMetadata.hasDeliverAtTime() ? msgMetadata.getDeliverAtTime() : -1L;
-            return delayedDeliveryTracker.get().addMessage(ledgerId, entryId, deliverAtTime);
+            try {
+                return delayedDeliveryTracker.get().addMessage(ledgerId, entryId, deliverAtTime);
+            } catch (OutOfDirectMemoryError ex) {
```

Comment:
In general, the Error type is not recommended to catch, we should use these two ways to solve the problem
- Set the maximum memory usage of the delay message. If the delay message usage exceeds the maximum memory usage, stuck new read
- we should kill the process if the error occurs

Reply:
The current implementation is the first approach, but we don't introduce a fixed value.  Instead, we use a dynamic value to do it.

> Set the maximum memory usage of the delay message.
This is hard to set a reasonable value.

> stuck new read
when we recover?

Reply:
Just like the implementation of JVM, if could not allocate more memory, the progress will be killed by OOM. It did not do `yield` and try another once.

>> Set the maximum memory usage of the delay message. If the delay message usage exceeds the maximum memory usage, stuck new read
> This is an ideal approach. what is the good time value to recover? 

In this patch, you mv this message into `replay queue`, and the next reading will get it again, loop. The result is actually the same as doing a stuck new read.